### PR TITLE
add experiment tags to MLFlow client

### PR DESCRIPTION
**Pull request by ankit-db**
## What changes are proposed in this pull request?
 
This is a PR for adding experiment tags to the MLFlow client. See #1630  for the sibling PR on the storage side. After discussion with Matei, we're going to go with a separated API, meaning we will have set_experiment_tag separate from the set_tag semantics for the run tags. This PR adds tests so that this workflow is well tested on R, Java, and Python. There is no fluent API at the moment per design doc.
 
## How is this patch tested?
 
Unit testing and manual testing.
 
### Is this a user-facing change? 

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.
 
We are adding the ability to add tags to your experiments in MLFlow! This will mean that you can add notes and other key-value tags to your experiments and see them via the API.

### What component(s) does this PR affect?
 
- [ ] UI
- [ ] CLI 
- [x] API 
- [x] REST-API 
- [ ] Examples 
- [x] Docs
- [x] Tracking
- [ ] Projects 
- [ ] Artifacts 
- [ ] Models 
- [ ] Scoring 
- [ ] Serving
- [x] R
- [x] Java
- [x] Python

### How should the PR be classified in the release notes? Choose one:
 
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [x] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: docs/source/R-api.rst [hunk 1]
```diff
@@ -1343,6 +1343,47 @@ Arguments
...
+---------
+
++-------------------------------+--------------------------------------+
+| Argument                      | Description                          |
++===============================+======================================+
+| ``key``                       | Name of the tag. Maximum size is 255 |
+|                               | bytes. This field is required.       |
++-------------------------------+--------------------------------------+
+| ``value``                     | String value of the tag being        |
+|                               | logged. Maximum size is 500 bytes.   |
```
[the remark this answers is not in the document]
Comment (dbczumar) [reply docs/source/R-api.rst]: Per discussion here: https://github.com/mlflow/mlflow/pull/1630#discussion_r305529624, this doesn't seem accurate for run tags stored in SQLAlchemy

**Review by dbczumar — COMMENTED** [review]
@ankitmathur-db I left a few comments, mostly nits

File: mlflow/R/mlflow/R/tracking-experiments.R [hunk 2]
```diff
@@ -46,6 +46,32 @@ mlflow_list_experiments <- function(view_type = c("ACTIVE_ONLY", "DELETED_ONLY",
     tibble::as_tibble()
 }
 
+#' Set Experiment Tag
+#'
+#' Sets a tag on an experiment. Tags are experiment metadata that can be updated.
+#'
```
[the remark this answers is not in the document]
Comment (ankitmathur-db) [reply mlflow/R/mlflow/R/tracking-experiments.R]: Fixed also for Get Experiment, which is where I'd gotten this from.

File: mlflow/R/mlflow/R/tracking-experiments.R [hunk 3]
```diff
@@ -46,12 +46,36 @@ mlflow_list_experiments <- function(view_type = c("ACTIVE_ONLY", "DELETED_ONLY",
...
+#' @param key Name of the tag. Maximum size is 255 bytes. This field is required.
+#' @param value String value of the tag being logged. Maximum size is 500 bytes. This field is required.
```
Comment (dbczumar) [inline mlflow/R/mlflow/R/tracking-experiments.R]: Let's update this to reflect limit changes. We can use the same string as the one in the proto.

File: mlflow/R/mlflow/R/tracking-experiments.R [hunk 4]
```diff
@@ -46,12 +46,36 @@ mlflow_list_experiments <- function(view_type = c("ACTIVE_ONLY", "DELETED_ONLY",
     tibble::as_tibble()
 }
 
+#' Set Experiment Tag
+#'
+#' Sets a tag on an experiment. Tags are experiment metadata that can be updated.
+#'
+#' @param key Name of the tag. Maximum size is 255 bytes. This field is required.
```
Comment (dbczumar) [inline mlflow/R/mlflow/R/tracking-experiments.R]: `255` > `250`?

File: mlflow/R/mlflow/R/tracking-experiments.R [hunk 5]
```diff
@@ -46,12 +46,36 @@ mlflow_list_experiments <- function(view_type = c("ACTIVE_ONLY", "DELETED_ONLY",
...
+#' @param value String value of the tag being logged. Maximum size is 500 bytes. This field is required.
+#' @param experiment_id Identifer to get an experiment.
```
Comment (dbczumar) [inline mlflow/R/mlflow/R/tracking-experiments.R]: `Identifier to get an experiment` >> `ID of the experiment`

**Review by dbczumar — APPROVED** [review]
LGTM after small comments are addressed. Thanks @ankitmathur-db!

**Pull request merged**